wallet/src/Pos/Wallet/Web/Methods/Payment.hs

```diff
@@ -214,3 +225,57 @@ sendMoney SendActions{..} passphrase moneySource dstDistr = do
 
     logDebug "sendMoney: constructing response"
     fst <$> constructCTx srcWallet srcWalletAddrs diff th
+
+
+-- | For each specified transaction, if it has been canceled, creates
+-- new transaction with correspondent destination address and money.
+-- This works as `mapM newPayment`, so completion may take a lot of time.
+--
+-- NOTE: Work of this function relies on how wallet create transactions
+-- currently!
+reformCanceledTxs
+    :: MonadWalletWebMode m
+    => SendActions m -> PassPhrase -> ReformCanceledTxsParams -> m [CTx]
+reformCanceledTxs sendActions passphrase params = testOnlyEndpoint $ do
+    ptxs <- getPendingTxs
+    let canceledTxs :: [(TxId, (CId Wal, PtxPoolInfo))]
+        canceledTxs = flip mapMaybe ptxs $ \PendingTx{..} ->
+            case _ptxCond of
+                PtxWontApply _ poolInfo -> Just (_ptxTxId, (_ptxWallet, poolInfo))
+                _                       -> Nothing
+    logDebug $ sformat ("List of all canceled transactions ever: "%listJson)
+         (map fst canceledTxs)
+
+    txsToRecreate <- case rctpBanned params of
+        [] -> pure canceledTxs
+        bannedCTxs -> do
+            bannedTxs <- mapM decodeCTypeOrFail bannedCTxs
+            let bannedTxsSet = S.fromList bannedTxs
+            pure $ filter ((`S.notMember` bannedTxsSet) . fst) canceledTxs
+
+    logDebug $ sformat ("Transactions selected for recreation: "%listJson)
+        (map fst txsToRecreate)
+
+    fmap catMaybes . forM txsToRecreate $ \(txId, (wid, txEntry)) ->
+        handleAll (txCreationHandler txId) $ do
+            let moneySource = WalletMoneySource wid
+                outputs = _txOutputs (_thTx txEntry)
+            TxOut addr coin <-
+                maybeThrow noExpectedInput (outputs ^? ix primaryDestIndex)
+            let outDistr = (encodeCType addr, coin)
+            ctx <- sendMoney sendActions passphrase moneySource (one outDistr)
+
+            newTxId <- decodeCTypeOrFail (ctId ctx)
+            logInfo $ sformat ("Successfully recreated transaction "%build
+                              %", new transaction: "%build)
+                txId newTxId
+            return (Just ctx)
+  where
+    -- for all current transactions we have change address as 0-th output
+    -- and
```

Comment:
...and? 😀

Reply:
Just to double-check: in the comment we say the change address is at 0-th index, but yet `primaryDestIndex` is set to `1`. Am I right in thinking the comment should be read as: "The outputs are two: the first one has index `0` and is the change address, the second is the real destination and it has index 1, which is what we use to recreate the Tx". Am I correct?

Reply:
Yep )